### ISSUE TITLE
remove vcruntime140.dll

### ISSUE THREAD

**Issue by shuffle2** [ufrisk/LeechCore#5]
The repo contains vcruntime140.dll, which I'm pretty sure is not supposed to be distributed by projects. Why don't you just set all projects to statically link against the runtime, remove the "ReleaseMT" build target, and remove the vcruntime140.dll from the repo? I've made the changes locally but dont have python headers installed to provide a full merge-ready PR. (but really, shouldn't be including build outputs in the repo, either...)

**Comment by shuffle2** [comment]
Ah, there are things like:
```c
* -- ppbDataOut =  ptr to receive function allocated output - must be LocalFree'd by caller!
```
...splitting alloc/free across dll boundaries is a definite no-no.

**Comment by ufrisk** [comment]
Thanks; you're right in that the redistributable runtime shouldn't be there. Most often it's already installed and in the case if it's not I'll just link to the redist package at microsoft.com

There is however a reason why I've elected to do runtime linking, since I compile quite a few binaries, not just in this project, the overall size impact of adding static linking will be substantial. I prefer to keep the size of the binaries down.

Having the build outputs in the project, while not very nice form-wise it's convenient. But if I start making releases more frequently (= more versions of the binaries) I'll move it to a separate downloadable file in the releases section.

And thanks for the comment about the LocalAlloc, I'll add a LeechCore_MemFree function that abstracts that part away in the next release.

**Comment by shuffle2** [comment]
> And thanks for the comment about the LocalAlloc, I'll add a LeechCore_MemFree function that abstracts that part away in the next release.

Thanks! It's better style, altho after looking over LeechCore more, I think it's only done with LocalAlloc/LocalFree currently, so not a huge deal (they will always be implemented in the same system lib, as opposed to crt allocator, which could be implemented in different libraries depending how you link). That said, dunno how/if this code works on non-Windows.

**Comment by ufrisk** [comment]
Yeah, as you mention it has no impact on Windows compiled code - and it's heavily used by my MemProcFS project (Windows only). PCILeech which also compiles for Linux don't make use of it so actually I don't know how it behaves.

Anyway, it's changed in my internal dev branch, I hope to push the fix together with some other changes next week or more likely the week after that.

**Comment by ufrisk** [comment]
I've looked more into the matter of `vcruntime140.dll` which I copied from `C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Redist\MSVC\14.22.27821\x64\Microsoft.VC142.CRT` from my Visual Studio installation.

According to the following redistribution documentation/agreement from MIcrosoft it's OK to redistribute this file. https://docs.microsoft.com/en-us/visualstudio/releases/2019/redistribution

To sum up, it's 100% as intended. I'll be adding a redist.txt file clarifying this to avoid future misunderstandings. Your other comments holds true though and I'll keep this issue open until I've fixed it all :)

**Comment by ufrisk** [comment]
binaries now moved into releases section - closing issue.
